### PR TITLE
Add R117 release post

### PR DESCRIPTION
**Pull request by lukeindykiewicz**
Blog post describing Release R117 Biskupin
https://github.com/snowplow/snowplow/pull/4218
https://github.com/snowplow/snowplow/milestone/172

To do:
- [x] update release notes link
- [x] update Discourse forum link
- [x] spot prices paragraph

### CODE REVIEW

**Comment by lukeindykiewicz** [comment]
@peel Could you add description for TLS changes, please.

**Comment by lukeindykiewicz** [comment]
@chuwy Could you add description for spot market changes, please?

File: _posts/2019-11-01-snowplow-r117-biskupin/index.md [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+tags: [snowplow, release]
+author: Łukasz Indykiewicz
+category: Releases
+permalink: /blog/2019/11/01/snowplow-r117-biskupin/
+---
+
+Snowplow release R117 is named after [Biskupin](https://en.wikipedia.org/wiki/Biskupin), which is "an archaeological site and a life-size model of a late Bronze Age fortified settlement in north-central Poland that also serves as a archaeological open-air museum".
+
+This release focuses on delivering small, but necessary improvements to Snowplow ecosystem:
+
```
Comment (yalisassoon) [inline _posts/2019-11-01-snowplow-r117-biskupin/index.md]: Maybe replace "necessary" with "important"?

File: _posts/2019-11-01-snowplow-r117-biskupin/index.md [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+4. [Additional event fingerprint hashing methods (SHA1, SHA256, SHA384, SHA512)](#additional-event-fingerprint-hashing-methods)
+5. [Support to the spot market for core instances](#support-to-the-spot-market-for-core-instances)
+
+The `bad row` release has been renamed to R118 - check [roadmap](#roadmap) for more details.
+
+## TLS port binding and certificate
+
+### TLS/SSL Certificate termination
+As an additional security measure it is now possible to terminate TLS/SSL connection directly within `scala-stream-collector` using a battle-tested [lightben/ssl-config](https://lightbend.github.io/ssl-config/index.html).
+We introduce several new configuration parameters in order to fit most workflows and configurations.
```
Comment (yalisassoon) [inline _posts/2019-11-01-snowplow-r117-biskupin/index.md]: Maybe update with "We have introduced several new configuration parameters in order to accommodate different workflows and configurations"

File: _posts/2019-11-01-snowplow-r117-biskupin/index.md [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+​
+### Default redirect endpoint disabling
+Another simple, but important security improvement is a possibility to disable the default event submission redirect. This allows users to disable the default endpoint in favour of custom user-defined url. 
+For example, following configuration will only allow redirects for custom-defined `/com.acme/redirect-me` endpoint, whereas the default `/r/tp2` will not be available.
+```hocon
+enableDefaultRedirect = false
+paths {
+  "/com.acme/redirect-me" = "/r/tp2"
+}
+```
```
Comment (yalisassoon) [inline _posts/2019-11-01-snowplow-r117-biskupin/index.md]: Does the above highlighting syntax work on our blog? I think we used to use 

{% highlight hocon %}
...
{% endhighlight %}

Not sure if the three backticks work - just checking you've tried to build the post locally to check?

File: _posts/2019-11-01-snowplow-r117-biskupin/index.md [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+* SHA1
+* SHA256
+* SHA384
+* SHA512
+
+## Support to the spot market for core instances
+
+Credits for this contribution goes to [miike](https://github.com/miike).
+
+> to be filled
```
Comment (yalisassoon) [inline _posts/2019-11-01-snowplow-r117-biskupin/index.md]: This still needs documenting?

Comment (lukeindykiewicz) [reply _posts/2019-11-01-snowplow-r117-biskupin/index.md]: yes

File: _posts/2019-11-01-snowplow-r117-biskupin/index.md (hunk 3 above)
Comment (lukeindykiewicz) [reply _posts/2019-11-01-snowplow-r117-biskupin/index.md]: I have just double checked it, so three backticks works, but there is no support for `hocon` (backticks nor the other syntax you mentioned), so I set up it as json. It isn't perfect, but looks good enough imho.

File: _posts/2019-11-01-snowplow-r117-biskupin/index.md [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+
+```
+aws:
+  emr:
+    jobflow:
+      core_instance_bid: 0.3
+```
+
+
+# Updated components:
```
Comment (chuwy) [inline _posts/2019-11-01-snowplow-r117-biskupin/index.md]: We don't use such paragraph. Instead we describe all changes that we think are important enough in human language and have a very concise and engineer-centric "Upgrade" section.

File: _posts/2019-11-01-snowplow-r117-biskupin/index.md [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+```
+Note, that for backwards-compatibility, the endpoint is still exposed by default.
+
+## Referer parser refreshment
+
+Referer parser is a library that contains list of possible referrer hosts. We keep maintenance of `0.3.x` branch with backporting new identified hosts into old `0.3.0` version. Several new hosts were identified and added to the [`referer.yml`](https://s3-eu-west-1.amazonaws.com/snowplow-hosted-assets/third-party/referer-parser/referers-latest.yml)
+
+## IPv6 anonymization
+
+This release adds support for anonymization of IPv6 addresses. `anonSegments` parameter is added to enrichment configuration. It describes how many segments should be anonymized. Some examples might be found in our test scenarios in `AnonIpEnrichmentSpec` in `3-enrich/scala-common-enrich/src/test/scala/com.snowplowanalytics.snowplow.enrich.common/enrichments/registry/AnonIpEnrichmentSpec.scala`.
```
[the remark this answers is not in the document]
Comment (lukeindykiewicz) [reply _posts/2019-11-01-snowplow-r117-biskupin/index.md]: I believe it's very useful to show actual example with the context. I would keep it like this, unless you really don't think it's necessary.

File: _posts/2019-11-01-snowplow-r117-biskupin/index.md [hunk 7]
```diff
@@ -0,0 +1,147 @@
...
+
+## Scala Common Enrich
+Includes changes: [Referer-parser](#referer-parser-refreshment), [IPv6 anonymization](#ipv6-anonymization) and [additional fingerpring hashing methods](#additional-event-fingerprint-hashing-methods)
+
+## EmrEtlRunner
+
+Includes changes: [Support to the spot market for core instances](#support-to-the-spot-market-for-core-instances)
+
+## Other components
+`EmrEtlRunner`, `Beam Enrich` and `Stream Enrich` includes new `Scala Common Enrich`
```
[the remark this answers is not in the document]
Comment (lukeindykiewicz) [reply _posts/2019-11-01-snowplow-r117-biskupin/index.md]: Great catch - thanks!

File: _posts/2019-11-01-snowplow-r117-biskupin/index.md (hunk 5 above)
Comment (lukeindykiewicz) [reply _posts/2019-11-01-snowplow-r117-biskupin/index.md]: I believe it's useful to know what components are upgraded. The changes are described in the paragraphs above. How do you want me to change this?

File: _posts/2019-11-01-snowplow-r117-biskupin/index.md [hunk 8]
```diff
@@ -0,0 +1,151 @@
...
+
+1. [TLS port binding and certificate](#tls-port-binding-and-certificate)
+2. [Referer parser refreshment](#referer-parser-refreshment)
+3. [IPv6 anonymization](#ipv6-anonymization)
+4. [Additional event fingerprint hashing methods (SHA1, SHA256, SHA384, SHA512)](#additional-event-fingerprint-hashing-methods)
+5. [Support to the spot market for core instances](#support-to-the-spot-market-for-core-instances)
+
+<!--more-->
+
+The bad row release has been renamed to R118 - check [roadmap](#roadmap) for more details.
```
Comment (benjben) [inline _posts/2019-11-01-snowplow-r117-biskupin/index.md]: Should be only in the roadmap section.

File: _posts/2019-11-01-snowplow-r117-biskupin/index.md [hunk 9]
```diff
@@ -0,0 +1,151 @@
...
+enableDefaultRedirect = false
+paths {
+  "/com.acme/redirect-me" = "/r/tp2"
+}
+```
+Note, that for backwards-compatibility, the endpoint is still exposed by default.
+
+## Referer parser refreshment
+
+Referer parser is a library that contains list of possible referrer hosts. We keep maintenance of `0.3.x` branch with backporting new identified hosts into old `0.3.0` version. Several new hosts were identified and added to the [`referer.yml`](https://s3-eu-west-1.amazonaws.com/snowplow-hosted-assets/third-party/referer-parser/referers-latest.yml)
```
Comment (benjben) [inline _posts/2019-11-01-snowplow-r117-biskupin/index.md]:
```suggestion
Referer parser is a library that contains a list of possible referer hosts. We keep maintenance of `0.3.x` branch with backporting new identified hosts into old `0.3.0` version. Several new hosts were identified and added to the [`referer.yml`](https://s3-eu-west-1.amazonaws.com/snowplow-hosted-assets/third-party/referer-parser/referers-latest.yml).
```

File: _posts/2019-11-01-snowplow-r117-biskupin/index.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (benjben) [reply _posts/2019-11-01-snowplow-r117-biskupin/index.md]: I agree with Anton, I think that we only want to show some configuration examples and the associated results after enrichment. Also maybe we should remind that `anonOctets` will still be used for IPv4.

[the remark this answers is not in the document]
Comment (benjben) [reply _posts/2019-11-01-snowplow-r117-biskupin/index.md]: @yalisassoon maybe we want to add that this is a first step in our will to fully support IPv6?

File: _posts/2019-11-01-snowplow-r117-biskupin/index.md [hunk 10]
```diff
@@ -0,0 +1,151 @@
...
+
+## IPv6 anonymization
+
+This release adds support for anonymization of IPv6 addresses. `anonSegments` parameter is added to enrichment configuration. It describes how many segments should be anonymized. Some examples might be found in our test scenarios in `AnonIpEnrichmentSpec` in `3-enrich/scala-common-enrich/src/test/scala/com.snowplowanalytics.snowplow.enrich.common/enrichments/registry/AnonIpEnrichmentSpec.scala`.
+
+The schema has been updated to version `1-0-1`:
```
Comment (benjben) [inline _posts/2019-11-01-snowplow-r117-biskupin/index.md]:
```suggestion
The schema for the configuration of the enrichment has been updated to version `1-0-1`:
```

File: _posts/2019-11-01-snowplow-r117-biskupin/index.md [hunk 11]
```diff
@@ -0,0 +1,151 @@
...
+
+Credits for this contribution goes to [miike](https://github.com/miike).
+
+From this version, besides MD5, new hashing algorithms are supported:
+
+* SHA1
+* SHA256
+* SHA384
+* SHA512
+
```
Comment (benjben) [inline _posts/2019-11-01-snowplow-r117-biskupin/index.md]: Would be nice to precise that the version of the configuration has been updated.

File: _posts/2019-11-01-snowplow-r117-biskupin/index.md (hunk 5 above)
Comment (benjben) [reply _posts/2019-11-01-snowplow-r117-biskupin/index.md]: We usually use an `Upgrading` section. You can find examples for the previous releases:
- [R116](https://snowplowanalytics.com/blog/2019/09/12/snowplow-r116-madara-rider/#upgrading)
- [R115](https://snowplowanalytics.com/blog/2019/07/18/snowplow-r115-sigiriya-emretlrunner/#upgrading)
- [R114](https://snowplowanalytics.com/blog/2019/05/20/snowplow-r114-polonnaruwa-enrichments/#upgrading)

File: _posts/2019-11-01-snowplow-r117-biskupin/index.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (lukeindykiewicz) [reply _posts/2019-11-01-snowplow-r117-biskupin/index.md]: Example removed, as suggested by @chuwy and @benjben

File: _posts/2019-11-01-snowplow-r117-biskupin/index.md (hunk 8 above)
Comment (lukeindykiewicz) [reply _posts/2019-11-01-snowplow-r117-biskupin/index.md]: Removed

File: _posts/2019-11-01-snowplow-r117-biskupin/index.md (hunk 9 above)
Comment (lukeindykiewicz) [reply _posts/2019-11-01-snowplow-r117-biskupin/index.md]: changed

File: _posts/2019-11-01-snowplow-r117-biskupin/index.md (hunk 10 above)
Comment (lukeindykiewicz) [reply _posts/2019-11-01-snowplow-r117-biskupin/index.md]: :+1:

File: _posts/2019-11-01-snowplow-r117-biskupin/index.md (hunk 5 above)
Comment (lukeindykiewicz) [reply _posts/2019-11-01-snowplow-r117-biskupin/index.md]: Section `Upgrading` added

File: _posts/2019-11-01-snowplow-r117-biskupin/index.md [hunk 12]
```diff
@@ -0,0 +1,209 @@
...
+```yaml
+aws:
+  emr:
+    jobflow:
+      core_instance_bid: 0.3
+```
+
+
+# Upgrading
```
Comment (benjben) [inline _posts/2019-11-01-snowplow-r117-biskupin/index.md]: I think that we're missing the list of the new versions of components that have been upgraded.

Comment (lukeindykiewicz) [reply _posts/2019-11-01-snowplow-r117-biskupin/index.md]: I just removed it, I thought you both told me that the list is not part of the standard release blog post sections. I'll add it back.

Comment (benjben) [reply _posts/2019-11-01-snowplow-r117-biskupin/index.md]: R116 [blog post](https://snowplowanalytics.com/blog/2019/09/12/snowplow-r116-madara-rider/#upgrading) contains link to new SSC artifact on Bintray  I think that you should also mention the Docker image
R115 [blog post](https://snowplowanalytics.com/blog/2019/07/18/snowplow-r115-sigiriya-emretlrunner/#upgrading) contains link to new EER on Bintray 
R114 [blog post](https://snowplowanalytics.com/blog/2019/05/20/snowplow-r114-polonnaruwa-enrichments/#upgrading) contains links to new artifacts 

Basically in the `Upgrading` section you need to put the links to the new versions of artifacts as well as the changes in the configuration of the components if any

Comment (lukeindykiewicz) [reply _posts/2019-11-01-snowplow-r117-biskupin/index.md]: I have added links to artifacts in the `Upgrading` section, also I have added the components that only contains snowplow-common-enrich bump.
Should `Spark Enrich` be pointing to `.jar` from hosted assets?

Comment (benjben) [reply _posts/2019-11-01-snowplow-r117-biskupin/index.md]: From R114 [blog post](https://snowplowanalytics.com/blog/2019/05/20/snowplow-r114-polonnaruwa-enrichments/#upgrading) :

Finally, if you are a batch pipeline user, a new Spark Enrich can be used by setting the new version in your EmrEtlRunner configuration:
```
enrich:
  version:
    spark_enrich: 1.18.0 # WAS 1.17.0
```
or directly make use of the new Spark Enrich available at:

`s3://snowplow-hosted-assets/3-enrich/spark-enrich/snowplow-spark-enrich-1.18.0.jar`

Comment (lukeindykiewicz) [reply _posts/2019-11-01-snowplow-r117-biskupin/index.md]: Thanks :+1:

File: _posts/2019-11-01-snowplow-r117-biskupin/index.md [hunk 13]
```diff
@@ -0,0 +1,247 @@
+---
+layout: post
+title-short: "Snowplow R117 release"
+title: "Snowplow R117 release"
+tags: [snowplow, release]
+author: Łukasz Indykiewicz
+category: Releases
+permalink: /blog/2019/11/01/snowplow-r117-biskupin/
```
Comment (chuwy) [inline _posts/2019-11-01-snowplow-r117-biskupin/index.md]: The date (here and in filename)

File: _posts/2019-11-01-snowplow-r117-biskupin/index.md [hunk 14]
```diff
@@ -0,0 +1,247 @@
...
+Snowplow release R117 is named after [Biskupin](https://en.wikipedia.org/wiki/Biskupin), which is "an archaeological site and a life-size model of a late Bronze Age fortified settlement in north-central Poland that also serves as a archaeological open-air museum".
+
+This release focuses on delivering small, but important improvements to Snowplow ecosystem:
+
+1. [TLS port binding and certificate](#tls-port-binding-and-certificate)
+2. [Referer parser refreshment](#referer-parser-refreshment)
+3. [IPv6 anonymization](#ipv6-anonymization)
+4. [Additional event fingerprint hashing methods (SHA1, SHA256, SHA384, SHA512)](#additional-event-fingerprint-hashing-methods)
+5. [Support to the spot market for core instances](#support-to-the-spot-market-for-core-instances)
+
```
Comment (chuwy) [inline _posts/2019-11-01-snowplow-r117-biskupin/index.md]: Roadmap and getting help should be in contents.

File: _posts/2019-11-01-snowplow-r117-biskupin/index.md [hunk 15]
```diff
@@ -0,0 +1,247 @@
...
+---
+
+Snowplow release R117 is named after [Biskupin](https://en.wikipedia.org/wiki/Biskupin), which is "an archaeological site and a life-size model of a late Bronze Age fortified settlement in north-central Poland that also serves as a archaeological open-air museum".
+
+This release focuses on delivering small, but important improvements to Snowplow ecosystem:
+
+1. [TLS port binding and certificate](#tls-port-binding-and-certificate)
+2. [Referer parser refreshment](#referer-parser-refreshment)
+3. [IPv6 anonymization](#ipv6-anonymization)
+4. [Additional event fingerprint hashing methods (SHA1, SHA256, SHA384, SHA512)](#additional-event-fingerprint-hashing-methods)
```
Comment (chuwy) [inline _posts/2019-11-01-snowplow-r117-biskupin/index.md]: " (SHA1, SHA256, SHA384, SHA512)" is not part of the heading.

File: _posts/2019-11-01-snowplow-r117-biskupin/index.md [hunk 16]
```diff
@@ -0,0 +1,247 @@
...
+
+Referer parser is a library that contains a list of possible referer hosts. We keep maintenance of `0.3.x` branch with backporting new identified hosts into old `0.3.0` version. Several new hosts were identified and added to the [`referer.yml`](https://s3-eu-west-1.amazonaws.com/snowplow-hosted-assets/third-party/referer-parser/referers-latest.yml)
+
+## IPv6 anonymization
+
+This release adds support for anonymization of IPv6 addresses. `anonSegments` parameter is added to enrichment configuration. It describes how many segments should be anonymized.
+
+The schema for the configuration of the enrichment has been updated to version `1-0-1`:
+
+```json
```
Comment (chuwy) [inline _posts/2019-11-01-snowplow-r117-biskupin/index.md]: Why we post the whole config file here, but only mention a SchemaVer for fingerpritn enrichment? I'd post full configs only in Upgrading section.

File: _posts/2019-11-01-snowplow-r117-biskupin/index.md [hunk 17]
```diff
@@ -0,0 +1,247 @@
...
+			"anonOctets": 1,
+			"anonSegments": 1
+		}
+	}
+}
+```
+
+## Additional event fingerprint hashing methods
+
+A new version of the Snowplow Common Enrich can be found on [Maven repository](https://mvnrepository.com/artifact/com.snowplowanalytics/snowplow-common-enrich_2.11/0.38.0)
```
Comment (chuwy) [inline _posts/2019-11-01-snowplow-r117-biskupin/index.md]: We don't need to duplicate this sentence for every enrichment.

File: _posts/2019-11-01-snowplow-r117-biskupin/index.md [hunk 18]
```diff
@@ -0,0 +1,247 @@
...
+      "excludeParameters": ["cv", "eid", "nuid", "stm"],
+      "hashAlgorithm": "SHA1"
+    }
+  }
+}
+```
+
+## Support to the spot market for core instances
+
+A new version of the EmrEtl Runner can be found on [our Bintray repository](https://bintray.com/snowplow/snowplow-generic/snowplow) under `r117-biskupin` version.
```
Comment (chuwy) [inline _posts/2019-11-01-snowplow-r117-biskupin/index.md]: EmrEtlRuuner.

File: _posts/2019-11-01-snowplow-r117-biskupin/index.md [hunk 19]
```diff
@@ -0,0 +1,247 @@
...
+
+## Stream Enrich 
+
+A new version of the Stream Enrich can be found on [our Docker Hub repository](https://hub.docker.com/r/snowplow/stream-enrich-kinesis/tags) under `0.22.0` tag.
+
+It contains the newest Snowplow Common Enrich.
+
+## Spark Enrich
+
+Batch pipelines hasn't been extensively tested as part of our plans to focus on real-time and deprecate batch processing.
```
Comment (chuwy) [inline _posts/2019-11-01-snowplow-r117-biskupin/index.md]: Can you add a link to the the batch pipeline deprecation RFC, just afraid somebody still doesn't know about that.

File: _posts/2019-11-01-snowplow-r117-biskupin/index.md (hunk 13 above)
Comment (lukeindykiewicz) [reply _posts/2019-11-01-snowplow-r117-biskupin/index.md]: Set to 12.02 - let's hope this is the date :)

File: _posts/2019-11-01-snowplow-r117-biskupin/index.md (hunk 14 above)
Comment (lukeindykiewicz) [reply _posts/2019-11-01-snowplow-r117-biskupin/index.md]: I also added `Upgrading`

File: _posts/2019-11-01-snowplow-r117-biskupin/index.md (hunk 17 above)
Comment (lukeindykiewicz) [reply _posts/2019-11-01-snowplow-r117-biskupin/index.md]: I thought it would be easier to find, but I would remove it.

**Comment by lukeindykiewicz** [comment]
@benjben the tsl/ssl config examples removed from description (left in Upgrading section)

**Comment by lukeindykiewicz** [comment]
The code for R117 is already released. @nepsil0n, could you check if everything is ok and merge the PR?
